File: src/other/format.kt
package other

fun main() {
    println(3.toBinaryStr())
    println(4.toBinaryStr())
    println(5.toBinaryStr())

    println()
    println()

    println(3.toOctalStr())
    println(4.toOctalStr())
    println(5.toOctalStr())
    println(8.toOctalStr())

    println()
    println()

    println(3.toHexString())
    println(4.toHexString())
    println(5.toHexString())
    println(8.toHexString())
    println(16.toHexString())
    println(32.toHexString())
}

// todo
fun Int.toBinaryStr() = this.toString(2)
fun Int.toOctalStr() = this.toString(8)
fun Int.toHexString() = this.toString(16)

//fun Int.toBinary(): String {
//    val builder = StringBuilder()
//    var temp = this
//
//    while (temp > 0) {
//        builder.insert(0, temp % 2)
//        temp = temp.shr(1)
//    }
//
//    return builder.toString()
//}
//
//fun Int.toOctal(): String {
//    val builder = StringBuilder()
//    var temp = this
//
//    while (temp > 0) {
//        builder.insert(0, temp % 8)
//        temp = temp.shr(3)
//    }
//
//    return builder.toString()
//}
//
//fun Int.toHex(): String {
//    val builder = StringBuilder()
//    var temp = this
//
//    while (temp > 0) {
//        builder.insert(0, temp % 16)
//        temp = temp.shr(4)
//    }
//
//    return builder.toString()
//}

File: lib/other/format.dart
void main() {
  print(1.toBinaryString());
  print(2.toBinaryString());
  print(3.toBinaryString());
  print(4.toBinaryString());
  print(5.toBinaryString());

  print("");
  print("");

  print(3.toOctalString());
  print(4.toOctalString());
  print(5.toOctalString());
  print(8.toOctalString());
  print(16.toOctalString());

  print("");
  print("");

  print(3.toHexString());
  print(4.toHexString());
  print(5.toHexString());
  print(8.toHexString());
  print(16.toHexString());
  print(32.toHexString());
}

/// todo
extension IntExtension on int {
  String toBinaryString() => this.toRadixString(2);

  String toOctalString() => this.toRadixString(8);

  String toHexString() => this.toRadixString(16);

  // String toBinaryString() {
  //   var str = "";
  //   var temp = this;
  //
  //   while (temp > 0) {
  //     str = "${temp % 2}$str";
  //     temp = temp >> 1;
  //   }
  //
  //   return str;
  // }
  //
  // String toOctalString() {
  //   var str = "";
  //   var temp = this;
  //
  //   while (temp > 0) {
  //     str = "${temp % 8}$str";
  //     temp = temp >> 3;
  //   }
  //
  //   return str;
  // }
  //
  // String toHexString() {
  //   var str = "";
  //   var temp = this;
  //
  //   while (temp > 0) {
  //     str = "${temp % 16}$str";
  //     temp = temp >> 4;
  //   }
  //
  //   return str;
  // }
}
